### PR TITLE
Implement breakpoint variable evaluation dynamic sizing

### PR DESCRIPTION
**Pull request by hxiong388**
Add a total maximum evaluation size on breakpoints (32KB). When breakpoint evaluation will happen normally until the total size of all the evaluated variables exceed the maximum limit, then further variable evaluations will return with a "buffer full" warning message.

Each variable evaluation now also has a size limit. User defined expressions have 32KB limit, while normal stack frame local variables only has 1KB limit.

### CODE REVIEW

File: google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb [hunk 1]
```diff
@@ -85,12 +85,21 @@ class Variable
...
+          MAX_MEMBERS = 1000
 
           ##
           # Max length on variable inspect results. Truncate extra and replace
           # with ellipsis.
-          MAX_STRING_LENGTH = 260
+          MAX_STRING_LENGTH = 500
+
+          ##
+          # @private Minimum amoutn of size limit needed for evaluation.
```
Comment (dazuma) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb]: s/amoutn/amount/

File: google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb [hunk 2]
```diff
@@ -150,6 +169,9 @@ def initialize
           #   {Google::Cloud::Debugger::Breakpoint::Variable::MAX_DEPTH}
           # @param [Breakpoint::VariableTable] var_table A variable table
           #   to store shared compound variables. Optional.
+          # @param [Integer] limit Maximum number of bytes this convertion
```
Comment (dazuma) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb]: s/convertion/conversion/
And again two lines down. And in fact, multiple times in the examples below.

File: google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb (hunk 1 above)
Comment (hxiong388) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb]: fixed

File: google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb (hunk 2 above)
Comment (hxiong388) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb]: conversion

File: google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb (hunk 1 above)
Comment (blowmage) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint/variable.rb]: I'm glad I'm not the only one who does this...

**Pull request merged**